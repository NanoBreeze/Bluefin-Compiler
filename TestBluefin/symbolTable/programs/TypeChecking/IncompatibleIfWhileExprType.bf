/*
Verify that the expr in if/while statements has bool type
*/

int main() {

	if (true) {} //ok

	while( (1 + 2) > 3) {} //ok

	if (1) {} // '1' does not have bool type

	while (5*6.3 - false) {} // expr doesn't have bool type, so show only expr's error, not while statement's error

	while ("a") {}

	return 0;
}