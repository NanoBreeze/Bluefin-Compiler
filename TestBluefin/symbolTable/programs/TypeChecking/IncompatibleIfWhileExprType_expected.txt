If statement. Expected: 'bool', Found: 'int'
Operator '-' cannot be applied to 'float' and 'bool'
While statement. Expected: 'bool', Found: 'string'
